pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -509,17 +518,24 @@ func (r *ReconcileClusterPool) createCluster(
 	}
 	logger.WithField("cluster", ns.Name).Info("Creating new cluster")
 
+	// Skip machinepools if we're provided a complete installConfigTemplate
+	// as the machinepool will replace the workers specified in the
+	// installconfigtemplate
+	skipMachinePools := installConfigTemplate != ""
```

Comment:
I think this assumption should be removed, we accepted that a user might give us a worker pool def in install config, and then change that work pool def day 2, obviously it's a rare situation where someone would want to do that but we considered it a non-goal to prevent it. 

However I think users with install-configs still might want to manage their workers from Hive day 2, so there's no reason to assume to skip machine pools just because you used your own template. 

This *might* be boiling down to a need for worker instance type control on the hiveutil CLI at some point.

Reply:
If I remember correctly, when I was looking through this section of the code, skipMachinePools isn't really exposed anywhere. 
So if a user specifies a clusterpool with installconfigsecrettemplate, there is no way to include in the spec that machinepools should be disabled. The implication of this is that the machinepool will use the hard-coded worker spec, which will overwrite the definition in the templated install-config.yaml

I think there are three potential approaches:
1. Disable the machinepool when using install config template (current implementation)
2. Expose skipMachinePools in clusterpool spec
3. Unmarshal install-config template sections to Hive MachinePoolPlatform and populate the relevant node info from the install-config template (probably requires completion and merging of #1182 + some additional work)

For our particular use case we specifically want the owner of the cluster (i.e. the user who claims it) to manage the workers, not Hive, but maybe that's an edge-case in the big picture.

Reply:
That is a perfectly reasonable use case. I don't think use of install config implies it, so I would rule out 1, and I think option 2 is my preference. Could be a followup PR.

Reply:
Exposing this would work fine for me. I'm happy to do it as a follow up PR, so that I can start with a clean base.

I'd be doing it fairly swiftly as the lack of ability to disable machinepools will render this feature noop for us, as hive will reconcile the workers down to a size that is smaller than what we need in our clusters.

What's the best way to proceed?

Reply:
I'd rather hold this PR and lump that change in, than put this assumption in, and someone start using it only to get surprised because it changes shortly thereafter. If it's disruptive to merge this without the assumption, lets lump it into this PR.

Reply:
It was a very straight forward change. Updated it now.